src/main/java/bio/terra/service/auth/iam/sam/SamIam.java

```diff
@@ -85,8 +85,14 @@ private ApiClient getApiClient(String accessToken) {
     ApiClient apiClient = new ApiClient();
     apiClient.setAccessToken(accessToken);
     apiClient.setUserAgent("OpenAPI-Generator/1.0.0 java"); // only logs an error in sam
-    apiClient.setConnectTimeout(
-        configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS));
+
+    // Sometimes Sam calls can take longer than the OkHttp default of 10 seconds to return a
+    // response.  In those cases, we can see socket timeout exceptions despite the underlying Sam
+    // call continuing to execute and possibly succeeding.
+    int operationTimeoutSeconds =
+        configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS);
+    apiClient.setReadTimeout(operationTimeoutSeconds * 1000);
```

Comment:
Instead of using `* 1000` I think it's better to use `Duration` to do this conversion. One way to write this is:
```suggestion
    long operationTimeout =        Duration.ofSeconds(configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS)).toMillis();
    apiClient.setReadTimeout((int) operationTimeout));
```

Reply:
Good idea.  I toyed with something very close to this, using the Duration but I missed the cast to int.